unit_tests/FileTests.d

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: Mar 12, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+import std.stdio;
+import Common;
+import dstep.translator.CodeBlock;
+import dstep.translator.Translator;
+
+unittest
+{
+    assertTranslates(
+q"EOS
+float foo(int x);
+float bar(int x);
+
+int a;
+EOS",
+q"/extern (C):
+
+float foo (int x);
+float bar (int x);
+extern __gshared int a;/");
+}
+
+unittest
+{
+    assertTranslatesObjCFile(
+        "test_files/objc/categories.d",
+        "test_files/objc/categories.h");
+}
+
+unittest
+{
+    assertTranslatesObjCFile(
+        "test_files/objc/classes.d",
+        "test_files/objc/classes.h");
+}
+
+unittest
+{
+    assertTranslatesObjCFile(
+        "test_files/objc/methods.d",
+        "test_files/objc/methods.h");
+}
+
+unittest
+{
+    assertTranslatesObjCFile(
+        "test_files/objc/primitives.d",
+        "test_files/objc/primitives.h");
+}
+
+unittest
+{
+    assertTranslatesObjCFile(
+        "test_files/objc/properties.d",
+        "test_files/objc/properties.h");
+}
+
+unittest
+{
+    assertTranslatesObjCFile(
+        "test_files/objc/protocols.d",
+        "test_files/objc/protocols.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/aggregate.d",
+        "test_files/aggregate.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/arrays.d",
+        "test_files/arrays.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/const.d",
+        "test_files/const.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/enums.d",
+        "test_files/enums.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/function_pointers.d",
+        "test_files/function_pointers.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/functions.d",
+        "test_files/functions.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/structs.d",
+        "test_files/structs.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/typedef.d",
+        "test_files/typedef.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/typedef_struct.d",
+        "test_files/typedef_struct.h");
+}
+
+unittest
+{
+    assertTranslatesCFile(
+        "test_files/unions.d",
+        "test_files/unions.h");
+}
```

Comment:
> I was thinking about completely replacing cucumber. What do you think about this?

Sure, go ahead, remove them. The tests that are Cucumber tests now are kind of pointless to have as Cucumber tests. But I know @Dicebot likes them :wink:.